python/across_api/base/api.py

```diff
@@ -19,3 +23,167 @@
     },
     root_path="/labs/api/v1",
 )
+
+# FastAPI Depends definitions
+
+
+# Depends functions for FastAPI calls.
+async def optional_daterange(
+    begin: Annotated[
+        Optional[datetime],
+        Query(
+            description="Start time of period to be calculated.",
+            title="Begin",
+        ),
+    ] = None,
+    end: Annotated[
+        Optional[datetime],
+        Query(
+            description="Start time of period to be calculated.",
+            title="End",
+        ),
+    ] = None,
+) -> dict:
+    """
+    Helper function to convert begin and end to datetime objects.
+    """
+    if begin is None or end is None:
+        return {"begin": None, "end": None}
+    return {"begin": Time(begin), "end": Time(end)}
+
+
+OptionalDateRangeDep = Annotated[dict, Depends(optional_daterange)]
+
+
+# Depends functions for FastAPI calls.
+async def optional_length(
+    length: Annotated[
+        Optional[float],
+        Query(
+            description="Length of time (days).",
+            title="Length",
+        ),
+    ] = None,
+) -> Optional[u.Quantity]:
+    """
+    Helper function to convert begin and end to datetime objects.
+    """
+    if length is None:
+        return None
+    return length * u.day
+
+
+OptionalLengthDep = Annotated[dict, Depends(optional_length)]
+
+
+async def optional_limit(
+    limit: Annotated[
+        Optional[int],
+        Query(
+            ge=0,
+            title="Limit",
+            description="Maximum number of results to return.",
+        ),
+    ] = None,
+) -> Optional[int]:
+    return limit
+
+
+LimitDep = Annotated[Optional[int], Depends(optional_limit)]
+
+
+async def error_radius(
+    error_radius: Annotated[
+        Optional[float],
+        Query(
+            ge=0,
+            title="Error Radius",
+            description="Error radius in degrees.",
+        ),
+    ] = None,
+) -> Optional[float]:
```

Comment:
Incorrect type.

```suggestion
) -> Optional[u.Quantity['angle']]:
```

Reply:
Fixed in [1a908cf](https://github.com/nasa-gcn/gcn.nasa.gov/pull/1975/commits/1a908cfc67a4771903e4aac26de1c5622bb97bd3). I'll note that mypy doesn't like `u.Quantity['angle']` so I used `u.Quantity[u.deg]`.